addons/mail/static/src/discuss/core/public/welcome_page.js

```diff
@@ -91,10 +97,43 @@ export class WelcomePage extends Component {
         }
     }
 
+    async onClickBlur() {
+        this.store.settings.toggleBlurSetting();
+        await this.applyBlur();
+    }
+
+    async applyBlur() {
+        if (!this.state.videoStream) {
+            return;
+        }
+        if (!this.store.settings.useBlur) {
+            this.videoRef.el.srcObject = this.state.videoStream;
+            return;
+        }
+        this.blurManager?.close();
+        this.blurManager = undefined;
+        try {
+            this.blurManager = new BlurManager(this.state.videoStream, {
+                backgroundBlur: this.store.settings.backgroundBlurAmount,
+                edgeBlur: this.store.settings.edgeBlurAmount,
+            });
+            this.videoRef.el.srcObject = await this.blurManager.stream;
+        } catch (_e) {
+            this.videoRef.el.srcObject = this.state.videoStream;
+            this.notification.add(
+                _t("%(name)s: %(message)s", { name: _e.name, message: _e.message }),
+                { type: "warning" }
+            );
+            this.store.settings.useBlur = false;
+        }
```

Comment:
This is copy-paste with rtc service use of the blur manager code.
Prefer reusing the code by splittting into a function that you simply code in each occurrences of code that needs the blur feature.